### PR TITLE
Fix PHPStan level 0 problems

### PR DESCRIPTION
**Pull request by szepeviktor**
@bonny Fixed many type problems to make static analysis possible.

### CODE REVIEW

File: dropins/class-rss-dropin.php [hunk 1]
```diff
@@ -174,15 +174,6 @@ public function check_for_rss_feed_request() {
...
 
-	/**
-	 * Modify capability check so all users reading rss feed (logged in or not) can read all loggers
-	 */
-	public function onCanReadSingleLogger( $user_can_read_logger, $logger_instance, $user_id ) {
-		$user_can_read_logger = true;
-
-		return $user_can_read_logger;
-	}
-
```
Comment (szepeviktor) [inline dropins/class-rss-dropin.php:185]: This was replaced by core's `__return_true`.

File: index.php [hunk 2]
```diff
@@ -33,44 +33,46 @@
 	die();
 }
 
-$ok_wp_version = version_compare( $GLOBALS['wp_version'], '6.1', '>=' );
-$ok_php_version = version_compare( phpversion(), '7.4', '>=' );
+if (
+	version_compare( phpversion(), '7.4', '<' )
+	|| version_compare( $GLOBALS['wp_version'], '6.1', '<' )
+) {
```
Comment (szepeviktor) [inline index.php:39]: Removed two globals.

File: index.php [hunk 3]
```diff
@@ -33,44 +33,46 @@
...
+	// User is running to old version of php, add admin notice about that.
+	require_once __DIR__ . '/inc/oldversions.php';
+	add_action( 'admin_notices', 'simple_history_old_version_admin_notice' );
 
-if ( $ok_php_version && $ok_wp_version ) {
```
Comment (szepeviktor) [inline index.php:39]: Removed `else` branch.

File: loggers/class-post-logger.php [hunk 4]
```diff
@@ -263,7 +263,7 @@ public function on_admin_action_editpost() {
 
 		$prev_post_data = get_post( $post_ID );
 
-		if ( is_wp_error( $prev_post_data ) ) {
+		if ( null === $prev_post_data ) {
```
Comment (szepeviktor) [inline loggers/class-post-logger.php:266]: `get_post` returns null on failure.
https://developer.wordpress.org/reference/functions/get_post/

File: phpstan.neon [hunk 5]
```diff
@@ -1,18 +1,15 @@
...
-    - %rootDir%/../../php-stubs/wp-cli-stubs/wp-cli-commands-stubs.php
-    - %rootDir%/../../php-stubs/wp-cli-stubs/wp-cli-i18n-stubs.php
-  ignoreErrors:
-    - '#Constant SIMPLE_HISTORY_PATH not found.#'
-    - '#Constant SIMPLE_HISTORY_DIR_URL not found.#'
-    - '#Constant WPINC not found.#'
-    - '#Constant WP_PLUGIN_DIR not found.#'
+    - vendor/php-stubs/wp-cli-stubs/wp-cli-stubs.php
+  # TODO Add stubs for 3rd-party loggers
+  level: 2
```
Comment (szepeviktor) [inline phpstan.neon:15]: @bonny You may continue on level 2. There will be much less bugs!

**Comment by bonny** [comment]
Thanks for this @szepeviktor. 

I saw that you removed this part from the phpstan config:

```
  ignoreErrors:
    - '#Constant SIMPLE_HISTORY_PATH not found.#'
    - '#Constant SIMPLE_HISTORY_DIR_URL not found.#'
    - '#Constant WPINC not found.#'
    - '#Constant WP_PLUGIN_DIR not found.#'
```

but without that I get a lot of constants not found when running level 1 or higher, for example
```
 ------ ---------------------------------------------------------------------
  Line   dropins/class-settings-debug-tab-dropin.php
 ------ ---------------------------------------------------------------------
  23     Constant SIMPLE_HISTORY_PATH not found.
```

I guess you do not get those errors since you removed them? Do you know any reason why I get those warning, like am I starting phpstan wrong or something?

**Comment by szepeviktor** [comment]
> Do you know any reason why I get those warning, like am I starting phpstan wrong or something?

No.
Noooo!

https://github.com/bonny/WordPress-Simple-History/blob/3f1dcd49ff50627b4ec0bb9e498b6a182b41fbc7/phpstan.neon#L14

There is a TODO to fix those.
You could add a bootstrap file `tests/phpstan/bootstrap.php`
```php
<?php

define('SIMPLE_HISTORY_PATH', './');
define('SIMPLE_HISTORY_DIR_URL', '/');
define('WPINC', 'inc');
define('WP_PLUGIN_DIR, 'plugins');
```
Values are ignored!
```neon
parameters:
    bootstrapFiles:
        - tests/phpstan/bootstrap.php
```

**Comment by bonny** [comment]
Ok thanks! I will do it that way then. I was confused that the phpstan docs said

> As of [PHPStan 1.7.0](https://github.com/phpstan/phpstan/releases/tag/1.7.0) nothing special is required to analyse code with global constants.

So I thought they should be found since they are added in `index.php`.

**Comment by szepeviktor** [comment]
Temporarily use literal values
https://github.com/bonny/WordPress-Simple-History/blob/3f1dcd49ff50627b4ec0bb9e498b6a182b41fbc7/index.php#L54-L56
I think PHPStan cannot use functions as values.

**Comment by szepeviktor** [comment]
In WordPress these are not constant but [configuration values](https://github.com/szepeviktor/starter-plugin/blob/9a1a8ffb5d2ef35266fd0f0c2bbda9ce931b6471/plugin-name.php#L77-L88).

**Pull request merged**